src/material-experimental/mdc-slider/slider.ts

```diff
@@ -87,67 +61,83 @@ export class MatSliderChange {
-  @Output() readonly change: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
+export class MatSlider implements OnDestroy, AfterViewInit {
+  @ViewChild('startThumb') startThumb: MatSliderThumb;
+  @ViewChild('endThumb') endThumb: MatSliderThumb;
 
-  /** Event emitted when the slider thumb moves. */
-  @Output() readonly input: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
+  @ViewChild('startInput') startInput: ElementRef<HTMLInputElement>;
+  @ViewChild('endInput') endInput: ElementRef<HTMLInputElement>;
 
-  /**
-   * Emits when the raw value of the slider changes. This is here primarily
-   * to facilitate the two-way binding for the `value` input.
-   * @docs-private
-   */
-  @Output() readonly valueChange: EventEmitter<number> = new EventEmitter<number>();
+  @ViewChild('trackActive') _trackActive: ElementRef<HTMLElement>;
+  get trackActive(): HTMLElement { return this._trackActive.nativeElement; }
 
-  /** Tabindex for the slider. */
-  @Input() tabIndex: number = 0;
+  // TODO(wagnermaciel): Throw a warning if this isn't set by the user.
+  @Input() ariaLabel: string | null;
 
-  /** The color palette for this slider. */
-  @Input() color: ThemePalette = 'accent';
+  /** The end value of a ranged slider. */
+  @Input()
+  get endValue(): number|null {
+    if (this._endValue === null) {
+      this.endValue = this.max;
+    }
+    return this._endValue;
+  }
+  set endValue(value: number|null) {
+    this._endValue = coerceNumberProperty(value);
+    this._changeDetectorRef.markForCheck();
+  }
+  private _endValue: number|null = null;
 
-  /**
-   * Function that will be used to format the value before it is displayed
-   * in the thumb label. Can be used to format very large number in order
-   * for them to fit into the slider thumb.
-   */
-  @Input() displayWith: (value: number) => string | number;
+  @Input()
+  get hasTickMarks(): boolean {
+    return this._hasTickMarks;
+  }
+  set hasTickMarks(value: boolean) {
+    // TODO(wagnermaciel): Warn user if they try to set hasTickMarks to true with isDiscrete false.
+    this._hasTickMarks = this.isDiscrete && coerceBooleanProperty(value);
+  }
+  private _hasTickMarks: boolean;
 
-  /** The minimum value that the slider can have. */
+  /** Whether the slider is disabled. */
   @Input()
-  get min(): number {
-    return this._min;
+  get isDisabled(): boolean {
+    return this._isDisabled;
   }
-  set min(value: number) {
-    this._min = coerceNumberProperty(value);
+  set isDisabled(value) {
+    this._isDisabled = coerceBooleanProperty(value);
   }
-  private _min = 0;
+  private _isDisabled = false;
+
+  /** Whether the slider is discrete. */
+  @Input()
+  get isDiscrete(): boolean {
+    return this._isDiscrete;
+  }
+  set isDiscrete(value) {
+    this._isDiscrete = coerceBooleanProperty(value);
+  }
+  private _isDiscrete = false;
+
+  /** Whether this is a ranged slider. */
+  @Input()
+  get isRange(): boolean {
```

Comment:
I thought we were going to have a design doc about this before adding new stuff to the API?

Reply:
We are. This current API that's implemented here is not the final one. This is as basic of an implementation I could create to get an understanding of how creating MDC components works